### PR TITLE
Add Debug information when e2e Test run Fails

### LINKED ISSUE #30: Improve debug information when e2e test run fail
Now we have some debug information when the e2e test run fails, but we need to improve and add more information that can be helpful to detect earlier the failure.

Current debug information:

- Istio CR Yaml
- Pods in Istio CR namespace
- Logs from sail-operator pod
- Logs from istiod pod

We need to add more debugging information as:

- Operator Pod YAML
- OC describe
- More logs if possible

### PR DESCRIPTION
**Pull request by fjglira**
Fixes #30

### CODE REVIEW

File: tests/e2e/util/common/e2e_utils.go [hunk 1]
```diff
@@ -90,28 +92,57 @@ func CheckNamespaceEmpty(ctx SpecContext, cl client.Client, ns string) {
...
+	operator, err := kubectl.GetYAML(namespace, "deployment", deploymentName)
 	if err != nil {
-		GinkgoWriter.Println("Error getting Istio CR: ", err)
+		GinkgoWriter.Println("Error getting operator deployment yaml: ", err)
 	}
-	GinkgoWriter.Println("Istio CR: \n", resource)
+	GinkgoWriter.Println("Operator deployment: \n", operator)
 
-	output, err := kubectl.GetPods(controlPlaneNamespace, "-o wide")
+	describe, err := kubectl.Describe(namespace, "deployment", deploymentName)
```
Comment (luksa) [inline tests/e2e/util/common/e2e_utils.go:105]: Why do we need both the YAML and the description? They both show the same information. I'd remove the description and leave just the YAML.

File: tests/e2e/util/kubectl/kubectl.go [hunk 2]
```diff
@@ -141,6 +141,21 @@ func GetPods(ns string, args ...string) (string, error) {
...
+func Describe(ns, kind, name string) (string, error) {
+	cmd := kubectl("describe %s %s %s", kind, name, nsflag(ns))
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return "", fmt.Errorf("error describing resource: %w, output: %s", err, output)
+	}
+
+	return output, nil
+}
+
```
Comment (luksa) [inline tests/e2e/util/kubectl/kubectl.go:158]: This is now unused code, which we should probably remove.

File: tests/e2e/util/common/e2e_utils.go [hunk 3]
```diff
@@ -90,7 +92,23 @@ func CheckNamespaceEmpty(ctx SpecContext, cl client.Client, ns string) {
 
 func LogDebugInfo() {
 	// General debugging information to help diagnose the failure
-	// TODO: Add more debugging information for others resources
+	// TODO: Add the creation of file with this information to be attached to the test report
+
```
Comment (nrfox) [inline tests/e2e/util/common/e2e_utils.go:96]: `kubectl events -n <namespace>` is also generally useful debug info as well.

Comment (fjglira) [reply tests/e2e/util/common/e2e_utils.go:96]: Thanks. I'll add it

**Review by luksa — APPROVED** [review]

**Pull request merged**